openerp/addons/base/tests/test_qweb.py

```diff
@@ -85,6 +86,12 @@ def __call__(self, name, options):
                 arch = etree.tostring(root, encoding='utf-8', xml_declaration=True)
                 return arch
 
+def stringify_children(node):
+    parts = ([node.text] +
+            list(chain(*([c.text, etree.tostring(c), c.tail] for c in node.getchildren() if not isinstance(c, etree._Comment)))) +
+            [node.tail])
+    # filter removes possible Nones in texts and tails
+    return ''.join(filter(None, parts))
```

Comment:
Also the `parts` thing should probably use chain.from_iterable and ifilter, there's no reason to reify a list until the final join call.